stm32f0_hal/src/pwm.rs

```diff
@@ -1,73 +1,190 @@
+                    tim3.ccer.modify(|_, w| w.cc2e().active());
+                }
+                Pin::PC6 => {
+                    gpioc.moder.modify(|_, w| w.moder6().alternate());
+                    gpioc.afrl.modify(|_, w| w.afrl6().af0());
+                    gpioc.otyper.modify(|_, w| w.ot6().push_pull());
+                    tim3.ccmr1_output
+                        .modify(|_, w| w.oc1m().pwmmode1().oc1pe().enabled());
+                    tim3.ccer.modify(|_, w| w.cc1p().clear_bit());
+                    tim3.ccer.modify(|_, w| w.cc1e().active());
+                }
+                Pin::PC7 => {
+                    gpioc.moder.modify(|_, w| w.moder7().alternate());
+                    gpioc.afrl.modify(|_, w| w.afrl7().af0());
+                    gpioc.otyper.modify(|_, w| w.ot7().push_pull());
+                    tim3.ccmr1_output
+                        .modify(|_, w| w.oc2m().pwmmode1().oc2pe().enabled());
+                    tim3.ccer.modify(|_, w| w.cc2p().clear_bit());
+                    tim3.ccer.modify(|_, w| w.cc2e().active());
+                }
+                Pin::PC8 => {
+                    gpioc.moder.modify(|_, w| w.moder8().alternate());
+                    gpioc.afrh.modify(|_, w| w.afrh8().af0());
+                    gpioc.otyper.modify(|_, w| w.ot8().push_pull());
+                    tim3.ccmr2_output
+                        .modify(|_, w| w.oc3m().pwmmode1().oc3pe().enabled());
+                    tim3.ccer.modify(|_, w| w.cc3p().clear_bit());
+                    tim3.ccer.modify(|_, w| w.cc3e().active());
+                }
+                Pin::PC9 => {
+                    gpioc.moder.modify(|_, w| w.moder9().alternate());
+                    gpioc.afrh.modify(|_, w| w.afrh9().af0());
+                    gpioc.otyper.modify(|_, w| w.ot9().push_pull());
+                    tim3.ccmr2_output
+                        .modify(|_, w| w.oc4m().pwmmode1().oc4pe().enabled());
+                    tim3.ccer.modify(|_, w| w.cc4p().clear_bit());
+                    tim3.ccer.modify(|_, w| w.cc4e().active());
+                }
+            }
 
-pub fn set_period(_period: u32) {
-    cortex_m::interrupt::free(|cs| {
-        let tim3 = TIM3.borrow(cs);
+            tim3.cr1.write(|w| {
+                w.cms()
+                    .center_align_mode1()
+                    .dir()
+                    .up()
+                    .opm()
+                    .continuous()
+                    .cen()
+                    .enabled()
+            });
 
-        // Set Prescaler Register - 16 bits
-        tim3.psc.write(|w| w.psc().bits(7));
+            tim3.egr.write(|w| w.ug().set_bit());
+        });
+        Pwm { pin }
+    }
 
-        // Set Auto-Reload register - 32 bits
-        let arr = 10000;
-        tim3.arr.write(|w| w.arr_h().bits((arr >> 16) as u16));
-        tim3.arr.write(|w| w.arr_l().bits(arr as u16));
-    });
-}
+    pub fn set_frequency(&self, frequency: u32) {
```

Comment:
If I understand correctly, this set_frequency actually impact all PWM instances? 
Maybe it's better to make it a class method and for the moment add a comment specifying that you should reset all duty of your instances. What do you think?